src/Core/Domain/Product/Command/UpdateProductSeoCommand.php

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * Updates Product SEO options
+ */
+class UpdateProductSeoCommand
+{
+    /**
+     * @var ProductId
+     */
+    private $productId;
+
+    /**
+     * @var string[]|null
+     */
+    private $localizedMetaTitles;
+
+    /**
+     * @var string[]|null
+     */
+    private $localizedMetaDescriptions;
+
+    /**
+     * @var string[]|null
+     */
+    private $localizedLinkRewrites;
+
+    /**
+     * @var ProductRedirectOption|null
+     */
+    private $redirectOption;
+
+    /**
+     * @param int $productId
+     */
+    public function __construct(int $productId)
+    {
+        $this->productId = new ProductId($productId);
+    }
+
+    /**
+     * @return ProductId
+     */
+    public function getProductId(): ProductId
+    {
+        return $this->productId;
+    }
+
+    /**
+     * @return string[]|null
+     */
+    public function getLocalizedMetaTitles(): ?array
+    {
+        return $this->localizedMetaTitles;
+    }
+
+    /**
+     * @param string[] $localizedMetaTitles key => value pairs where each key represents language id
+     *
+     * @return UpdateProductSeoCommand
+     */
+    public function setLocalizedMetaTitles(array $localizedMetaTitles): self
+    {
+        $this->localizedMetaTitles = $localizedMetaTitles;
+
+        return $this;
+    }
+
+    /**
+     * @return string[]|null
+     */
+    public function getLocalizedMetaDescriptions(): ?array
+    {
+        return $this->localizedMetaDescriptions;
+    }
+
+    /**
+     * @param string[] $localizedMetaDescriptions key => value pairs where each key represents language id
+     *
+     * @return UpdateProductSeoCommand
+     */
+    public function setLocalizedMetaDescriptions(array $localizedMetaDescriptions): self
+    {
+        $this->localizedMetaDescriptions = $localizedMetaDescriptions;
+
+        return $this;
+    }
+
+    /**
+     * @return string[]|null
+     */
+    public function getLocalizedLinkRewrites(): ?array
+    {
+        return $this->localizedLinkRewrites;
+    }
+
+    /**
+     * @param string[] $localizedLinkRewrites key => value pairs where each key represents language id
+     *
+     * @return UpdateProductSeoCommand
+     */
+    public function setLocalizedLinkRewrites(array $localizedLinkRewrites): self
+    {
+        $this->localizedLinkRewrites = $localizedLinkRewrites;
+
+        return $this;
+    }
+
+    /**
+     * @return ProductRedirectOption|null
+     */
+    public function getRedirectOption(): ?ProductRedirectOption
+    {
+        return $this->redirectOption;
+    }
+
+    /**
+     * @param string $redirectType
+     * @param int $redirectTargetId
+     *
+     * @return UpdateProductSeoCommand
+     */
+    public function setRedirectOption(string $redirectType, int $redirectTargetId): self
+    {
+        if ($redirectType === ProductRedirectOption::TYPE_NO_REDIRECT) {
+            $this->redirectOption = ProductRedirectOption::buildNoRedirect();
```

Comment:
Since @eternoendless recently advised not to use this kind of factory methods I would suggest an alternative that is easy to use as well Create a `ProductNoRedirectOption` class that extends `ProductRedirectOption` and forces the preset values for no redirect

Reply:
Indeed. @zuk3975 please see https://github.com/PrestaShop/PrestaShop/pull/20554 that introduces a `NoManufacturerId` . This is something we heavily discussed and agreed on: no more "zero value" or "no value" but rather "zero class" or "no class".